applications: serial_lte_modem: add LwM2M Carrier Settings support (#13165)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  * ``#XMQTTCFG`` AT command to configure MQTT client before connecting to the broker.
  * The :ref:`CONFIG_SLM_AUTO_CONNECT <CONFIG_SLM_AUTO_CONNECT>` Kconfig option to support automatic LTE connection at start-up or reset.
  * The :ref:`CONFIG_SLM_CUSTOMER_VERSION <CONFIG_SLM_CUSTOMER_VERSION>` Kconfig option for customers to define their own version string after customization.

>   * ``#XCARRIERCFG`` AT command to configure the LwM2M carrier library by means of the LwM2M carrier settings (see :kconfig:option:`CONFIG_LWM2M_CARRIER_SETTINGS`).

Reviewer: ```suggestion
  * ``#XCARRIERCFG`` AT command to configure the LwM2M carrier library using the LwM2M carrier settings (see the :kconfig:option:`CONFIG_LWM2M_CARRIER_SETTINGS` Kconfig option).
```
